server/src/test/java/org/elasticsearch/search/aggregations/pipeline/movfn/MovFnWhitelistedFunctionTests.java

```diff
@@ -313,6 +313,10 @@ public void testEmptySimpleStdDev() {
         assertThat(actual, equalTo(Double.NaN));
     }
 
+    public void testStdDevNaNAvg() {
+        assertThat(MovingFunctions.stdDev(new double[] { 1.0, 2.0, 3.0 }, Double.NaN), equalTo(Double.NaN));
```

Comment:
Not sure if this test is useful since it cannot check the early termination of the method, but might be worth adding just to cover that code path, wdyt?

Reply:
++ can't hurt :)